tests/test_histogramming.py

```diff
@@ -13,3 +16,82 @@ def test_make_hist():
     assert h.sum().variance == 5
     assert h[0].value == 1
     assert h[-1].value == 1
+
+
+def test_upper_edge_inclusive():
+    """
+    Check that upper egde of last bin is inclusive by default.
+    This comes from the fact that make_hist() turns off the underflow and overflow bins by default.
+    """
+
+    data = [0] * 3 + [1] * 4 + [2] * 5 + [3] * 6 + [4] * 7 + [5] * 8
+    h1 = make_hist(data=data * 2, bins=5, range=(0, 5))
+    h2 = make_hist(data=data * 2, bins=[0, 1, 2, 3, 4, 5])
+    h3 = make_hist(data=data, bins=[0, 1, 2, 3, 4, 5], weights=2)
+    h4 = make_hist(data=data, bins=[0, 1, 2, 3, 4, 5], weights=[2] * len(data))
+
+    expected = [6, 8, 10, 12, 30]
+
+    assert np.array_equal(h1.values(), expected)
+    assert np.array_equal(h2.values(), expected)
+    assert np.array_equal(h3.values(), expected)
+    assert np.array_equal(h4.values(), expected)
+
+
+def test_upper_edge_exclusive():
+    """
+    Check that upper egde of last bin is exclusive when overflow=True.
+    """
+
+    data = [0] * 3 + [1] * 4 + [2] * 5 + [3] * 6 + [4] * 7 + [5] * 8
+
+    a1 = create_axis(5, (0, 5), overflow=True)
+    a2 = create_axis([0, 1, 2, 3, 4, 5], overflow=True)
+    a3 = create_axis(5, data=data, overflow=True)
+    # For a integer axis, the last bin is always exclusive, no matter the overflow setting.
+    int_a1 = bh.axis.Integer(0, 5, overflow=False)
+    int_a2 = bh.axis.Integer(0, 5, overflow=True)
+
+    h1 = bh.Histogram(a1, storage=bh.storage.Weight())
+    h2 = bh.Histogram(a2, storage=bh.storage.Weight())
+    h3 = bh.Histogram(a3, storage=bh.storage.Weight())
+    h4 = bh.Histogram(int_a1, storage=bh.storage.Weight())
+    h5 = bh.Histogram(int_a2, storage=bh.storage.Weight())
+    h6 = h1.copy()
+    h7 = h2.copy()
+    h8 = h3.copy()
+    h9 = h4.copy()
+    h10 = h5.copy()
+
+    h1.fill(data * 2)
+    h2.fill(data * 2)
+    h3.fill(data * 2)
+    h4.fill(data * 2)
+    h5.fill(data * 2)
+    h6.fill(data, weight=[2] * len(data))
+    h7.fill(data, weight=[2] * len(data))
+    h8.fill(data, weight=[2] * len(data))
+    h9.fill(data, weight=[2] * len(data))
+    h10.fill(data, weight=[2] * len(data))
+
+    expected = [6, 8, 10, 12, 14]
+
+    assert np.array_equal(h1.values(), expected)
+    assert np.array_equal(h2.values(), expected)
+    assert np.array_equal(h3.values(), expected)
+    assert np.array_equal(h4.values(), expected)
+    assert np.array_equal(h5.values(), expected)
+    assert np.array_equal(h6.values(), expected)
+    assert np.array_equal(h7.values(), expected)
+    assert np.array_equal(h8.values(), expected)
+    assert np.array_equal(h9.values(), expected)
+    assert np.array_equal(h10.values(), expected)
+
+
+def test_range_coverage_warning():
```

Comment:
Could we also have a check with a weighted histogram (two cases: a simple `int` and a `list`)?